### PR TITLE
libdeflate: add version 1.14

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **libdeflate/***

partial support conan v2.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libdeflate/all/conanfile.py [hunk 1]
```diff
@@ -1,5 +1,9 @@
-from conans import ConanFile, AutoToolsBuildEnvironment, VisualStudioBuildEnvironment, tools
+from conan import ConanFile
 from conan.tools.microsoft import is_msvc
+from conan.tools.files import export_conandata_patches, apply_conandata_patches, get, chdir, rmdir, copy, rm
+from conan.tools.env import Environment
+from conans import MSBuild, AutoToolsBuildEnvironment, VisualStudioBuildEnvironment
+from conans.tools import vcvars, environment_append
 import os
 
 required_conan_version = ">=1.33.0"
```
Comment (jwillikers) [inline recipes/libdeflate/all/conanfile.py:9]:
```suggestion
required_conan_version = ">=1.52.0"
```

**Review by jwillikers — CHANGES_REQUESTED** [review]

**Review by jwillikers — DISMISSED** [review]

File: recipes/libdeflate/all/conandata.yml [hunk 2]
```diff
@@ -15,18 +18,33 @@ sources:
     url: "https://github.com/ebiggers/libdeflate/archive/refs/tags/v1.7.tar.gz"
     sha256: "a5e6a0a9ab69f40f0f59332106532ca76918977a974e7004977a9498e3f11350"
 patches:
+  "1.14":
+    - patch_file: "patches/1.14-0001-fix-makefiles.patch"
+      base_path: "source_subfolder"
+      patch_description: "disable optimization and apply compiler settings on conan recipe"
+      patch_type: "backport"
```
Comment (prince-chrismc) [inline recipes/libdeflate/all/conandata.yml:25]:
```suggestion
      patch_type: "conan"
```

File: recipes/libdeflate/all/conandata.yml [hunk 3]
```diff
@@ -15,18 +18,33 @@ sources:
...
   "1.12":
     - patch_file: "patches/1.12-0001-fix-makefiles.patch"
       base_path: "source_subfolder"
+      patch_description: "disable optimization and apply compiler settings on conan recipe"
+      patch_type: "backport"
```
Comment (prince-chrismc) [inline recipes/libdeflate/all/conandata.yml:30]:
```suggestion
      patch_type: "conan"
```

File: recipes/libdeflate/all/conandata.yml [hunk 4]
```diff
@@ -15,18 +18,33 @@ sources:
...
   "1.10":
     - patch_file: "patches/1.9-0001-fix-makefiles.patch"
       base_path: "source_subfolder"
+      patch_description: "disable optimization and apply compiler settings on conan recipe"
+      patch_type: "backport"
```
Comment (prince-chrismc) [inline recipes/libdeflate/all/conandata.yml:35]:
```suggestion
      patch_type: "conan"
```

File: recipes/libdeflate/all/conandata.yml [hunk 5]
```diff
@@ -15,18 +18,33 @@ sources:
...
   "1.9":
     - patch_file: "patches/1.9-0001-fix-makefiles.patch"
       base_path: "source_subfolder"
+      patch_description: "disable optimization and apply compiler settings on conan recipe"
+      patch_type: "backport"
```
Comment (prince-chrismc) [inline recipes/libdeflate/all/conandata.yml:40]:
```suggestion
      patch_type: "conan"
```

File: recipes/libdeflate/all/conandata.yml [hunk 6]
```diff
@@ -15,18 +18,33 @@ sources:
...
   "1.8":
     - patch_file: "patches/1.7-0001-fix-makefiles.patch"
       base_path: "source_subfolder"
+      patch_description: "disable optimization and apply compiler settings on conan recipe"
+      patch_type: "backport"
```
Comment (prince-chrismc) [inline recipes/libdeflate/all/conandata.yml:45]:
```suggestion
      patch_type: "conan"
```

File: recipes/libdeflate/all/conandata.yml [hunk 7]
```diff
@@ -15,18 +18,33 @@ sources:
...
   "1.7":
     - patch_file: "patches/1.7-0001-fix-makefiles.patch"
       base_path: "source_subfolder"
+      patch_description: "disable optimization and apply compiler settings on conan recipe"
+      patch_type: "backport"
```
Comment (prince-chrismc) [inline recipes/libdeflate/all/conandata.yml:50]:
```suggestion
      patch_type: "conan"
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
The backport type is for source code changes that are really takign fixes from upstream...

Please use that one the least amount possible. 

This is really a "conanize" since it's just allow conan to pass in it's settings/flags

***

let me know if the docs are unclear about what each one means so I can improve them

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libdeflate/all/conanfile.py [hunk 8]
```diff
@@ -35,43 +39,48 @@ def _settings_build(self):
...
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not is_msvc(self) and \
-           not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        if self._settings_build.os == "Windows" and not is_msvc(self):
+            if "CONAN_BASH_PATH" not in Environment().vars(self, scope="build").keys():
```
Comment (SpaceIm) [inline recipes/libdeflate/all/conanfile.py:65]: `Environment().vars(self, scope="build").keys()` is not equivalent to `tools.get_env("CONAN_BASH_PATH")`

Comment (uilianries) [reply recipes/libdeflate/all/conanfile.py:65]: True!

**Pull request merged**